src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.Windows.cs

```diff
@@ -299,8 +308,64 @@ public ApartmentState GetApartmentState()
             }
         }
 
-        // TODO: https://github.com/dotnet/corefx/issues/20766
-        public bool TrySetApartmentState(ApartmentState state) { throw new PlatformNotSupportedException(); }
+        public bool TrySetApartmentState(ApartmentState state)
+        {
+            if (this != CurrentThread)
+            {
+                if (HasStarted())
+                    throw new ThreadStateException();
+                _initialAppartmentState = state;
```

Comment:
Is this value (i.e. stored in _initialAppartmentState) supposed to be used for initialization when the target thread starts?

Reply:
Good catch

Reply:
There is also a potential race condition with starting the thread.  In Priority setter we wrapped the code with `using (LockHolder.Hold(_lock))` to prevent the race.  Note that internal locking behavior in these classes is already quite different comparing to desktop CLR.